src/tests/playwright/pages/util/recorder/recorder-processor.js

```diff
@@ -0,0 +1,43 @@
+const MAX_RENDER_QUANTUM = 128;
+
+/**
+ * @classdesc The AudioWorklet RecorderProcessor records raw PCM samples
+ * from node input into a fixed-length recording buffer. Data is posted once
+ * the buffer is filled. Will also pass through audio from input to output.
+ */
+class RecorderProcessor extends AudioWorkletProcessor {
+  constructor(options) {
+    super();
+    this._position = 0;
+    this._maxSamples = options.processorOptions.maxSamples;
+    this._recordingBuffer = new Array(this.numberOfChannels)
```

Comment:
buffer - AudioBuffer
array - ChannelData

I think this concept works consistently across many WebAudio tests.